lib/_stream_writable.js

```diff
@@ -741,9 +740,8 @@ ObjectDefineProperties(Writable.prototype, {
   writable: {
     get() {
       const w = this._writableState;
-      if (!w) return false;
-      if (w.writable !== undefined) return w.writable && !w.ended;
-      return Boolean(!w.destroyed && !w.errored && !w.ending);
+      return !!w && w.writable !== false && !w.destroyed && !w.errored &&
+        !w.ending;
```

Comment:
only `w.writable === false` has special meaning

Reply:
we might want to document this in a comment